AmplifyPlugins/DataStore/AWSDataStoreCategoryPlugin/Storage/SQLite/StorageEngineAdapter+SQLite.swift

```diff
@@ -295,6 +314,33 @@ final class SQLiteStorageEngineAdapter: StorageEngineAdapter {
         }
         completion(.successfulVoid)
     }
+
+    static func clearIfNewVersion(version: String,
+                                  dbFilePath: URL,
+                                  userDefaults: UserDefaults = UserDefaults.standard,
+                                  fileManager: FileManager = FileManager.default) throws {
+
+        guard let previousVersion = userDefaults.string(forKey: dbVersionKey) else {
+            return
+        }
+
+        if previousVersion == version {
+            return
+        }
+
+        guard fileManager.fileExists(atPath: dbFilePath.path) else {
+            return
+        }
+
+        log.verbose("\(#function) Warning: Schema change detected, removing your previous database")
+        do {
+            try fileManager.removeItem(at: dbFilePath)
+        } catch {
+            log.error("\(#function) Failed to delete database file located at: \(dbFilePath), error: \(error)")
+            throw error
```

Comment:
wrap in `DataStoreError.invalidDatabase(path: dbFilePath.absoluteString, error)`

Reply:
Done